### PR TITLE
Fixed data section init expr not i64

### LINKED ISSUE #17: [interpreter] Wrong typing of data segment
The interpreter still outputs i32.const init expressions for the data segment. This makes e.g. `memory64.wast` fail in v8: https://github.com/WebAssembly/memory64/blob/master/test/core/memory64.wast#L11

I don't know much OCaml, but one location which certainly needs fixing is `check_data` in `valid.ml`: https://github.com/WebAssembly/memory64/blob/master/interpreter/valid/valid.ml#L421

And in encoding, we probably need to introduce a `vec64` here: https://github.com/WebAssembly/memory64/blob/master/interpreter/binary/encode.ml#L501

### PR DESCRIPTION
**Pull request by aardappel**
Fixes: https://github.com/WebAssembly/memory64/issues/17

### CODE REVIEW

**Review by backes — APPROVED** [review]
Thanks for fixing! LGTM from my side, but I am not an OCaml expert. Please wait for a second approval.

**Comment by backes** [comment]
I just tested locally; this CL seems to make V8 pass all spec tests.

**Comment by aardappel** [comment]
@rossberg any comments?

**Comment by aardappel** [comment]
Ok, since this works for me and @backes I can probably merge this for now, any improvements can happen later.

**Review by rossberg — COMMENTED** [review]
Sorry, I was OOO. LGTM.

**Pull request merged**
